src/engine/UIManager.cxx

```diff
@@ -288,30 +282,16 @@ void UIManager::drawUI() const
 
 void UIManager::toggleGroupVisibility(const std::string &groupID, UIElement *sender)
 {
-  if (groupID.empty())
-  {
-    LOG(LOG_ERROR) << "UI Element with Action \"ToggleVisibilityOfGroup\" was called without a parameter.";
-    return;
-  }
-
   if (sender)
   {
     Button *button = dynamic_cast<Button *>(sender);
-    // cast the object to a Button to check if it's a toggle button.
-    if (button && button->getUiElementData().isToggleButton)
-    {
-      for (const auto &it : m_uiGroups[groupID])
```

Comment:
I'm not sure why this one statement has been removed, it doesn't really seem to be part of this branch. In anycase, this loop sets other buttons in this group to have the same visibility as the button that was pressed. Like this, all the buttons are either visible or invisible, rather than just this group. The way this works isn't great but I think that we'll need to keep this loop in here for now, this entire class will have to be refactored in the future and hopefully that makes it a bit more straight forward.

Reply:
Hmm you're right. I thought we had duplicated code but I was wrong